src/overlays/actors/ovl_En_Sth2/z_en_sth2.c

```diff
@@ -27,22 +30,95 @@ const ActorInit En_Sth2_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+#include "overlays/ovl_En_Sth2/ovl_En_Sth2.c"
+
+void EnSth2_Init(Actor* thisx, PlayState* play) {
+    EnSth2* this = THIS;
+
+    this->objIndex = Object_GetIndex(&play->objectCtx, OBJECT_STH);
+    Actor_SetScale(&this->actor, 0.01f);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 36.0f);
+    this->unused = 0;
+
+    if (play->actorCtx.unk5 & 2) {
+        this->actor.flags |= (ACTOR_FLAG_10 | ACTOR_FLAG_20);
+    } else {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    this->actionFunc = EnSth2_UpdateSkelAnime;
+}
+
+void EnSth2_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void EnSth2_UpdateSkelAnime(EnSth2* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+}
+
+void EnSth2_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnSth2* this = THIS;
+
+    if (Object_IsLoaded(&play->objectCtx, this->objIndex)) {
+        this->actor.objBankIndex = this->objIndex;
+        Actor_SetObjectDependency(play, &this->actor);
+        SkelAnime_InitFlex(play, &this->skelAnime, &object_sth_Skel_0031F8, &gEnSth2WavingHandAnim, this->jointTable,
+                           this->morphTable, OBJECT_STH_LIMB_MAX);
+        Animation_PlayLoop(&this->skelAnime, &gEnSth2WavingHandAnim);
+        this->actor.update = EnSth2_UpdateActionFunc;
+        this->actor.draw = EnSth2_Draw;
+    }
+}
+
+void EnSth2_UpdateActionFunc(Actor* thisx, PlayState* play) {
+    EnSth2* this = THIS;
+
+    this->actionFunc(this, play);
+}
+
+s32 EnSth2_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    s32 pad;
+
+    if (limbIndex == OBJECT_STH_LIMB_0F) {
+        *dList = gEnSth2HeadDL;
+    }
+    if ((limbIndex == OBJECT_STH_LIMB_08) || (limbIndex == OBJECT_STH_LIMB_0A) || (limbIndex == OBJECT_STH_LIMB_0D)) {
+        rot->y += (s16)(Math_SinS((play->state.frames * ((limbIndex * 50) + 0x814))) * 200.0f);
+        rot->z += (s16)(Math_CosS((play->state.frames * ((limbIndex * 50) + 0x940))) * 200.0f);
+    }
+    return false;
+}
 
-extern UNK_TYPE D_060031F8;
+void EnSth2_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    static Vec3f D_80BFA010 = { 700.0f, 400.0f, 0.0f };
```

Comment:
This is a focus offset position. The only thing I noticed in this file not named. Mind giving it one? Thankyou!